daras_ai_v2/urls.py

```
@@ -44,6 +45,11 @@ def paginate_queryset(
         try:
             # filter the queryset based on previous cursor
             qs = qs.filter(**{param: cursor[param]})
+        except ValidationError:
+            # error: URL param contains invalid value for a cursor param
+            # redirect to the first page (remove all ordering-related query params)
+            query_params = {k: v for k, v in cursor.items() if k not in ordering}
+            raise gui.QueryParamsRedirectException(query_params)
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)